conans/client/importer.py

```diff
@@ -96,29 +96,42 @@ def remove_imports(conanfile, copied_files, output):
                 output.warn("Unable to remove imported file from build: %s" % f)
 
 
-def run_deploy(conanfile, install_folder, output):
+def default_deploy(conanfile, deploy_folder):
+    conanfile.copy(pattern="*", dst=deploy_folder)
+
+
+def run_deploy(conanfile, install_folder, deploy_folder, output):
     deploy_output = ScopedOutput("%s deploy()" % output.scope, output)
     file_importer = _FileImporter(conanfile, install_folder)
     package_copied = set()
 
-    # This is necessary to capture FileCopier full destination paths
-    # Maybe could be improved in FileCopier
-    def file_copier(*args, **kwargs):
-        file_copy = FileCopier(conanfile.package_folder, install_folder)
-        copied = file_copy(*args, **kwargs)
-        _make_files_writable(copied)
-        package_copied.update(copied)
-
-    conanfile.copy_deps = file_importer
-    conanfile.copy = file_copier
-    conanfile.install_folder = install_folder
-    with get_env_context_manager(conanfile):
-        with tools.chdir(install_folder):
-            conanfile.deploy()
-
-    copied_files = file_importer.copied_files
-    copied_files.update(package_copied)
-    _report_save_manifest(copied_files, deploy_output, install_folder, "deploy_manifest.txt")
+    if (hasattr(conanfile, "deploy") and callable(conanfile.deploy)) or deploy_folder:
```

Comment:
In your current implementation, `deploy_folder` will always have a value, so it will always run the default deploy.